confgenerator/config.go

```diff
@@ -812,27 +825,38 @@ func (uc *UnifiedConfig) Validate(ctx context.Context) error {
 	return nil
 }
 
-func (l *Logging) Validate() error {
+func (uc *UnifiedConfig) ValidateLogging() error {
+	l := uc.Logging
 	subagent := "logging"
 	if len(l.Exporters) > 0 {
 		log.Print(`The "logging.exporters" field is no longer needed and will be ignored. This does not change any functionality. Please remove it from your configuration.`)
 	}
 	if l.Service == nil {
 		return nil
 	}
+	validReceivers := map[string]bool{}
+	for k := range l.Receivers {
+		validReceivers[k] = true
+	}
+	if uc.Combined != nil {
```

Comment:
Why do we have a helper for `HasCombined` that just does this? It's even more characters than the `!=nil` check, and it's used in exactly one place right now.

(BTW, I prefer seeing an explicit `!= nil` here because that's actually testing what is important - we need the `uc.Combined.Receivers` on the next line to not hit a nil-pointer dereference. `HasCombined` might have more complicated logic that doesn't necessarily correlate with `nil`.)

Reply:
`HasCombined` was created to be consistent with `HasLogging` & `HasMetrics`.  

We might want to consider removing all 3 methods or, make sure we are using it consistently.

Reply:
It looks like the `Has` versions of all three are barely being used; `cmd/ops_agent_windows/run_windows.go` is incorrectly using them to mean "has a non-empty logging/metrics section", while the implementions just check for the section existing at all (even if there are zero receivers or pipelines). The only other place they're used is `confgenerator/feature_tracking.go` which is using them in lieu of `nil` checks.

I'd rather not put that in this PR since it's already so large, but I think all three could be cleaned up and that would be a net improvement.